### PR TITLE
Bug-ID: CLOUDSTACK-8882: calculate network offering usage per nic instead of per network offering

### PR DESCRIPTION
**Pull request by kishankavala**
Issues occurs when:
A Vm is created with mutiple nics:
If 2 networks use same network offering, network offering usage will be 48hrs (assuming 24hrs aggregation)

Usage should be reported per Nic instead of network offering

Added test case to verify

### CODE REVIEW

**Comment by runseb** [comment]
Hi @kishankavala can we run some simulator tests to check this ?

**Comment by DaanHoogland** [comment]
@kishankavala in your test case you are creating two UsageNetworkOfferings and then assure that two are persisted. How does this test two nics being created from the same NetworkOffering to be counted seperately? The use case/bug you describe makes sense but I see no relation to the test. I would like to include this in 4.7 can you please add to it?

File: engine/schema/src/com/cloud/usage/UsageVO.java [hunk 1]
```diff
@@ -125,6 +125,25 @@ public UsageVO(Long zoneId, Long accountId, Long domainId, String description, S
...
+        this.vmInstanceId = vmId;
+        this.offeringId = offeringId;
+        this.usageId = usageId;
+        this.startDate = startDate;
+        this.endDate = endDate;
+        this.networkId = networkId;
+    }
+
+
     public UsageVO(Long zoneId, Long accountId, Long domainId, String description, String usageDisplay, int usageType, Double rawUsage, Long vmId, String vmName,
```
Comment (borisroman) [inline engine/schema/src/com/cloud/usage/UsageVO.java]: Isn't it possible to extends this one instead of creating a new constructor?

**Comment by rhtyd** [comment]
@kishankavala please rebase against latest master and push -f, update on status of your PR

tag:easypr

**Comment by bvbharatk** [comment]
### ACS CI BVT Run
 **Sumarry:**
 Build Number 114
 Hypervisor xenserver
 NetworkType Advanced
 Passed=73
 Failed=0
 Skipped=3

_Link to logs Folder (search by build_no):_ https://www.dropbox.com/sh/yj3wnzbceo9uef2/AAB6u-Iap-xztdm6jHX9SjPja?dl=0


**Failed tests:**

**Skipped tests:**
test_vm_nic_adapter_vmxnet3
test_static_role_account_acls
test_deploy_vgpu_enabled_vm

**Passed test suits:**
test_deploy_vm_with_userdata.py
test_affinity_groups_projects.py
test_portable_publicip.py
test_vpc_vpn.py
test_over_provisioning.py
test_global_settings.py
test_scale_vm.py
test_service_offerings.py
test_routers_iptables_default_policy.py
test_routers.py
test_reset_vm_on_reboot.py
test_snapshots.py
test_deploy_vms_with_varied_deploymentplanners.py
test_login.py
test_list_ids_parameter.py
test_public_ip_range.py
test_multipleips_per_nic.py
test_regions.py
test_affinity_groups.py
test_network_acl.py
test_pvlan.py
test_volumes.py
test_nic.py
test_deploy_vm_root_resize.py
test_resource_detail.py
test_secondary_storage.py
test_vm_life_cycle.py
test_disk_offerings.py

**Comment by kishankavala** [comment]
@DaanHoogland , Test creates two UsageNetworkOfferings, both with same network offering but different nicIds. Test case is verified by checking that 2 usage records are created instead of one. Without this fix only 1 usage record is created.

@rhtyd rebased against latest master

**Comment by cloudmonger** [comment]
### ACS CI BVT Run
 **Sumarry:**
 Build Number 369
 Hypervisor xenserver
 NetworkType Advanced
 Passed=104
 Failed=1
 Skipped=7

_Link to logs Folder (search by build_no):_ https://www.dropbox.com/sh/yj3wnzbceo9uef2/AAB6u-Iap-xztdm6jHX9SjPja?dl=0


**Failed tests:**
* test_routers_network_ops.py

 * test_01_RVR_Network_FW_PF_SSH_default_routes_egress_true Failed


**Skipped tests:**
test_01_test_vm_volume_snapshot
test_vm_nic_adapter_vmxnet3
test_static_role_account_acls
test_11_ss_nfs_version_on_ssvm
test_nested_virtualization_vmware
test_3d_gpu_support
test_deploy_vgpu_enabled_vm

**Passed test suits:**
test_deploy_vm_with_userdata.py
test_affinity_groups_projects.py
test_portable_publicip.py
test_over_provisioning.py
test_global_settings.py
test_scale_vm.py
test_service_offerings.py
test_routers_iptables_default_policy.py
test_loadbalance.py
test_routers.py
test_reset_vm_on_reboot.py
test_deploy_vms_with_varied_deploymentplanners.py
test_network.py
test_router_dns.py
test_non_contigiousvlan.py
test_login.py
test_deploy_vm_iso.py
test_list_ids_parameter.py
test_public_ip_range.py
test_multipleips_per_nic.py
test_regions.py
test_affinity_groups.py
test_network_acl.py
test_pvlan.py
test_volumes.py
test_nic.py
test_deploy_vm_root_resize.py
test_resource_detail.py
test_secondary_storage.py
test_vm_life_cycle.py
test_disk_offerings.py

**Comment by DaanHoogland** [comment]
@kishankavala i forgot about this one, guess we won't make 4.7 ;)

can you appease cloudmonger?

**Comment by kishankavala** [comment]
@DaanHoogland cloudmonger failures are not related to this PR. Changes in this PR are related to usage server only.

File: engine/schema/src/com/cloud/usage/UsageVO.java (hunk 1 above)
Comment (DaanHoogland) [inline engine/schema/src/com/cloud/usage/UsageVO.java]: @kishankavala can you answer @borisroman ?

**Comment by DaanHoogland** [comment]
@kishankavala please rebase and re-open if still relevant. closing

**Pull request closed without merging**